src/hooks/useDropdownOptions.ts

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+export interface Item {
+  [key: string]: string | number;
+}
+
+/**
+ * Transforms your array of objects into the shape our form system expects.
+ * Our form system expects an Array of Objects with a shape of [{label: String, value: String}]
+ * @param {string} label
+ * @param {string} value
+ * @param {array} items [{item}]
+ */
+export function useDropdownOptions(
+  label: string,
+  value: string,
+  items?: Item[]
+): {label: string | number; value: string | number | null}[] {
+  return React.useMemo(() => {
+    if (items && !isObjectArray(items)) {
+      throw Error(
+        'The items provided to useDropdownOptions must be an array of objects'
+      );
+    }
+
+    if (!items || !items.length) {
+      return [{label: '- -', value: null}];
+    }
+
+    return items.map((item) => ({label: item[label], value: item[value]}));
+  }, [items, label, value]);
+}
+
+const isObjectArray = (array: Object[]) => {
```

Comment:
I like inline types when they are short and sweet like this one. 

We'll have to put together a TypeScript style guide sometime in the future once we all settle on our preferences.